bowling/src/fileReader.cpp

```diff
@@ -0,0 +1,66 @@
+#include <algorithm>
+#include <fstream>
+
+#include "fileReader.hpp"
+
+namespace fs = std::experimental::filesystem;
+
+FileReader::FileReader(const std::string& directory) : directoryPath_(fs::path(directory)), isValid_(false) {
+    
+    if (!checkDirectory()) {
+        return;
+    }
+
+    std::vector<fs::path> files = makeFileList();
+    readFiles(files);
+    isValid_ = true;
+};
+
+bool FileReader::checkDirectory()
+{
+    if (!fs::exists(directoryPath_)) {
+        return false;
+    }
+    if (!fs::is_directory(directoryPath_)) {
+        return false;
+    }
+    return true;
+}
+
+std::vector<fs::path> FileReader::makeFileList() {
+    std::vector<fs::path> files;
+
+    for (const auto& entry : fs::directory_iterator(directoryPath_)) {
+        auto filename = entry.path().filename();
+        if (fs::is_regular_file(entry.status())) {
```

Comment:
As we already have `checkDirectory()` method, maybe we can use the same idea and create `checkFile()` method to check if given file is valid or not?

Reply:
come on, guys :)

Reply:
@m-kop My idea was that `checkFile()` would allow you to check not only if it "is_regular_file" but also you can additionally check the file extension if we would like to use only .txt files. Or any other file validation if needed.